### ISSUE TITLE
MySQL errors when recording both property.real, property.imag, writes complex when only recording property.real

### ISSUE THREAD

**Issue by blthayer** [gridlab-d/gridlab-d#1063]
MySQL module recorder objects do not seem to be capable of recording real and imaginary components separately. 

Here's a triplex_load with a nested recorder:
~~~~
object triplex_load {
  name "nd_sx3197624b";
  phases BS;
  nominal_voltage 120.09;
  base_power_1 zipload_schedule.value * 1005.15;
  base_power_2 zipload_schedule.value * 1005.15;
  power_pf_1 0.97;
  power_fraction_1 1.00;
  power_pf_2 0.97;
  power_fraction_2 1.00;
  object mysql.recorder {
	property "voltage_12.real", "voltage_12.imag";
	table "triplex_0";
	interval 60;
	options PURGE;
	header_fieldnames "name";
  };
}
~~~~
When running the model, the following errors occur, indicating that MySQL attempts to create duplicate column names:
~~~~
ERROR    [INIT] : init_recorder(obj=4993;unnamed): database:0: database:0->query[CREATE TABLE IF NOT EXISTS `triplex_0` (`id` INT AUTO_INCREMENT PRIMARY KEY, `t` TIMESTAMP, `voltage_12` CHAR(40), `voltage_12` CHAR(40), name CHAR(64), index i_name (name), INDEX `i_t` (`t`) )] failed - Duplicate column name 'voltage_12'
ERROR    [INIT] : init_by_deferral(): object recorder:4993 initialization failed
ERROR    [INIT] : model initialization failed
FATAL    [INIT] : shutdown after simulation stopped prematurely
FATAL    [INIT] : environment startup failed: Invalid argument
~~~~

**Additionally:** Attempting to record just property.value results in the full complex number being recorded in the database. The mysql recorder simply doesn't seem to extract .real or .imag.

**Comment by dchassin** [comment]
It looks like you cannot use separate columns for real and imaginary parts.

**Comment by blthayer** [comment]
@dchassin - shouldn't MySQL make a different column for the different sub-parts of the property? Like 'voltage_12_real' and 'voltage_12_imag'?

**Comment by dchassin** [comment]
The property name parse cannot handle parts in mysql/recorder.cpp:init().  This cannot be fixed easily until the problems in part name handling are fixed.

**Comment by ftuffner** [comment]
Sidebar discussions indicate another solution may be the "global complex output" typing/flag that has been discussed in the past.  Something to force all recorders to output in a common notation, not the mix of rectangular/polar that seems to happen.

**Comment by dchassin** [comment]
We probably should leave this open because it will require a minor mod in recorder.cpp to correctly use the part handling when it is fixed.

**Comment by eberleim** [comment]
This bug has been fixed by #1098 so I'm going to close it.
